draft-specifications/Model.md

```diff
@@ -0,0 +1,324 @@
+> - **Python**: `()` (We have not discussed whether to use tuple or list. Tuple
+>   is less mutable, though enforcing immutability in Python is not likely to be
+>   a goal the way it is in hardened JavaScript.)
+>
+> Lists participate in the JSON subset of OCapN.
+>
+> We achieved consensus to name the type "List" at the [November 14, 2023](https://github.com/ocapn/ocapn/issues/94) meeting.
+
+## Struct (JSON)
+
+A struct with unique, unordered string keys and values of heterogeneous type.
+
+> - **Guile**: `make-tbd-hash` a hash of undecided type
+> - **JavaScript**: `{}`
+> - **Python**: `{}`
+>
+> For the purposes of surviving a round trip, the order of appearance of
+> entries in the struct must not be important for determining equivalence.
+> A struct representation concretely using one key order may validly
+> round-trip into a struct representation using another key order.
+> However, we encourage concrete representations to use some canonical key
+> order, though this is not a concern of the abstract syntax and data model.
+>
+> A JavaScript object that owns any symbol-keyed properties or with a prototype
+> other than `Object.prototype` cannot be passed as an OCapN struct, as all such
+> objects are reserved for concrete representations of OCapN types.
+> The key `Symbol.for('passStyle')` is special and indicates the kind of OCapN
+> value represented by an object.
+
+## Tagged
+
+A value with a string tag.
+
+> OCapN provides a small number of container types with minimal semantics.
+> Tags allow protocols to emerge from values annotated to indicate a richer
+> variety of semantics.
+> Concretely, a list might be differentiated from or hinted to be an ordered
+> list, sorted list, or a set of unique values.
+>
+> - **Guile**: currently a Guile record labeled `'(syrec ...)` (**imported**)
+> - **JavaScript**: an object with the key `Symbol.for('passStyle')` and value
+>   `tagged`, with a string tagName and a single `"payload"` field carrying the tagged value.
+>   ```js
+>   {
+>     [Symbol.for('passStyle')]: 'tagged',
+>     [Symbol.toStringTag]: tagName,
+>     payload
+>   }
+>   ```
+> - **Python**: `Tagged(tag, value)` where `tag` is a `string` and `Tagged` is
+>   imported from `ocapn`.
+>
+> OCapN tagged data have a single value (are not variadic) [November 14, 2023](https://github.com/ocapn/ocapn/issues/94)
+>
+> What are tagged values for? https://github.com/ocapn/ocapn/issues/52
+
+## Remotable
+
+A value that can receive messages.
+
+A remotable value is either locally implemented or forwards messages to the peer
+where such a local implementation exists for delivery to it.
+The latter is referred to as a presence.
+
+Comparing the identities of remotable values is pass-invariant, meaning that
+comparing the identities of any pair of remotable values will produce the same
+result on one peer if passed over OCapN to another peer and compared there,
+regardless of whether either value is a presence on either peer.
```

Comment:
As I'm reviewing this now, I have trouble wrapping my head around this sentence but that could just be the different terminology (compared to Goblins) and how these are implemented in JS. Not sure this is useful feedback but I thought I'd flag it anyway.

Reply:
I’ve rewritten this section. Please let me know if it continue to be opaque.